web/src/main/java/cz/muni/fi/pa165/team/match/MatchController.java

```diff
@@ -0,0 +1,156 @@
+
+    private final TeamRepository teamRepository;
+
+    private final TeamMatchGoalRepository teamMatchGoalRepository;
+
+    private final TeamPlayerRepository teamPlayerRepository;
+
+    @Autowired
+    public MatchController(
+        TeamMatchFacade teamMatchFacade,
+        TeamMatchRepository teamMatchRepository,
+        TeamRepository teamRepository,
+        TeamMatchGoalRepository teamMatchGoalRepository,
+        TeamPlayerRepository teamPlayerRepository)
+    {
+        this.teamMatchFacade = teamMatchFacade;
+        this.teamMatchRepository = teamMatchRepository;
+        this.teamRepository = teamRepository;
+        this.teamMatchGoalRepository = teamMatchGoalRepository;
+        this.teamPlayerRepository = teamPlayerRepository;
+    }
+
+    @InitBinder("matchRequest")
+    private void initBinder(WebDataBinder binder)
+    {
+        binder.setValidator(new MatchCreationValidator());
+    }
+
+    @RequestMapping(value = "/matches", method = RequestMethod.GET)
+    public ModelAndView viewListPlayed()
+    {
+        return new ModelAndView("team/match/played")
+            .addObject("matchResults", teamMatchFacade.getPlayedMatchesList())
+            .addObject("formatter", DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+    }
+
+    @RequestMapping(value = "/matches/upcomming", method = RequestMethod.GET)
+    public ModelAndView viewListUpcomming()
+    {
+        return new ModelAndView("team/match/upcomming")
+            .addObject("plannedMatches", teamMatchRepository.findAllPlannedMatches())
+            .addObject("formatter", DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+    }
+
+    @RequestMapping(value = "/match/create", method = RequestMethod.GET)
+    public ModelAndView viewNewMatch()
+    {
+        return new ModelAndView("team/match/create")
+            .addObject("teams", teamRepository.findAll())
+            .addObject("matchRequest", new MatchRequest());
+    }
+
+    @RequestMapping(value = "/match/create", method = RequestMethod.POST)
+    public ModelAndView submitNewMatch(
+        @ModelAttribute("matchRequest") @Validated MatchRequest matchRequest,
+        BindingResult result
+    ) throws Exception
+    {
+        if (result.hasErrors()) {
+            return new ModelAndView("team/match/create")
+                .addObject("teams", teamRepository.findAll())
+                .addObject("matchRequest", matchRequest);
+        }
+
+        LocalDateTime startTime = getMVCLocalDateTime(matchRequest.getStartTime());
+        LocalDateTime endTime = getMVCLocalDateTime(matchRequest.getEndTime());
+
+        TeamMatch teamMatch = teamMatchFacade.createMatch(
+            UUID.fromString(matchRequest.getHomeTeamId()),
+            UUID.fromString(matchRequest.getAwayTeamId()),
+            startTime,
+            endTime
+        );
+
+        return new RedirectResponse("/match/" + teamMatch.getId().toString());
+    }
+
+    private LocalDateTime getMVCLocalDateTime(String formInputDate)
```

Comment:
pokud to nepůjde, udělal bych statický helper, ktery bych volal v getterech request objektu, aby rovnou vracely správný typ a nemuselo se to převádět v controllerech